crates/uv/tests/it/sync.rs

```diff
@@ -1226,6 +1226,54 @@ fn sync_dev_group() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn sync_non_existent_group() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = ["typing-extensions"]
+
+        [dependency-groups]
+        foo = []
+        bar = ["requests"]
+        "#,
+    )?;
+
+    context.lock().assert().success();
+
+    // Requesting a non-existent group should fail.
+    uv_snapshot!(context.filters(), context.sync().arg("--group").arg("baz"), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 7 packages in [TIME]
+    error: Could not find dependency group `baz`
```

Comment:
Curious that we fail _after_ resolution?

Reply:
We validate the requested groups during the sync operation rather than the lock operation. I guess we could do both.

Reply:
Might be annoying to do a long resolution then fail because the group doesn't exist.